service/lib/agama/http/clients/software.rb

```diff
@@ -77,19 +76,18 @@ def get_resolvables(unique_id, type, optional)
           JSON.parse(get("software/resolvables/#{unique_id}?type=#{type}&optional=#{optional}"))
         end
 
-        def provisions_selected?(_provisions)
-          # TODO: implement it, not sure how it should look like
-          []
+        def provisions_selected?(provisions)
+          provisions.select do |prov|
```

Comment:
In the D-Bus client, it says
```rb
        # @return [Array<Boolean>] An array containing whether each tag is selected or not
```
https://github.com/agama-project/agama/blob/v17/service/lib/agama/dbus/clients/software.rb#L107
```suggestion
          provisions.map do |prov|
```

Reply:
well, we use it only in one case for Autologin and in fact it will work for both cases :)
https://github.com/agama-project/agama/blob/master/service/lib/agama/dbus/y2dir/modules/Autologin.rb#L179
so it just calls `any?` on result. Maybe we can even change ruby code there?

Reply:
You have transposed an API from D-Bus to HTTP while changing its semantics and dropping the docs comment.
The fact that it continues to work is an accident.
Please add a docs comment